File: Max-number-of-k-sum-pairs.cpp
class Solution {
public:
    int maxOperations(vector<int>& nums, int k) {
        int count =0;
        sort(nums.begin(),nums.end());
        int i=0;
        int j=nums.size()-1;
        while(i<j){
            if(nums[i]+nums[j]<k){
                i++;
            }else  if(nums[i]+nums[j]>k){
                j--;
            }else{
                count++;
                i++;
                j--;
            }
        }
        return count;
    }
};

File: leet-code-solutions/max-number-of-k-sum-pairs.py
class Solution:
    def maxOperations(self, nums: List[int], k: int) -> int:
        nums.sort()
        c = 0
        l =0
        N = len(nums)
        r = N-1

        while l<r:
            if nums[l] + nums[r] > k:
                r-=1
            elif nums[l] + nums[r] < k:
                l+=1
            else:
                c+=1
                r-=1
                l+=1
        return c        